crates/context_aware_config/src/api/default_config/handlers.rs

```diff
@@ -184,3 +194,60 @@ async fn get(db_conn: DbConnection) -> superposition::Result<Json<Vec<DefaultCon
     let result: Vec<DefaultConfig> = default_configs.get_results(&mut conn)?;
     Ok(Json(result))
 }
+
+pub fn get_key_usage_context_ids(
+    key: &str,
+    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+) -> superposition::Result<Vec<String>> {
+    let result: Vec<Context> = contexts.load(conn).map_err(|err| {
+        log::error!("failed to fetch contexts with error: {}", err);
+        db_error!(err)
+    })?;
+
+    let mut context_ids = vec![];
+    for context in result.iter() {
+        from_value::<Map<String, Value>>(context.override_.to_owned())
+            .map_err(|err| {
+                log::error!("failed decode override into object: {}", err);
+                unexpected_error!("failed to decode override")
+            })?
+            .get(key)
+            .map_or((), |_| context_ids.push(context.id.to_owned()))
+    }
+    Ok(context_ids)
+}
+
+#[delete("/{key}")]
+async fn delete_default_config(
+    path: Path<String>,
+    db_conn: DbConnection,
+    user: User,
+) -> superposition::Result<HttpResponse> {
+    let DbConnection(mut conn) = db_conn;
+
+    let key = path.into_inner();
+    fetch_default_key(&key, &mut conn)?;
+    let context_ids = get_key_usage_context_ids(&key, &mut conn)
+        .map_err(|e| unexpected_error!("Something went wrong"))?;
+    if context_ids.is_empty() {
+        let deleted_row =
+            delete(default_configs.filter(db::schema::default_configs::key.eq(&key)))
```

Comment:
```suggestion
            diesel::delete(default_configs.filter(db::schema::default_configs::key.eq(&key)))
```

Now we can rename the function to just `delete` instead of `delete_default_config`.

Reply:
done